lib/puppet/transaction/persistence.rb

```diff
@@ -0,0 +1,81 @@
+require 'yaml'
+require 'puppet/util/yaml'
+
+# A persistence store implementation for storing information between
+# transaction runs for the purposes of information inference (such
+# as calculating corrective_change).
+# @api private
+class Puppet::Transaction::Persistence
```

Comment:
The last run report can't be extended to meet your needs I presume (I'm not entirely familiar with the difference b/w this new YAML file and what we already have)